src/components/TodoList.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useContext } from 'react';
+import { TodoItem } from './TodoItem';
+import { TodosContext } from './TodosContext';
+import * as todoService from '../api/todos';
+import { Todo } from '../types/Todo';
+import { FilterOption } from '../types/FilterOption';
+
+export const TodoList: React.FC = () => {
+  const { todos, filterOption, setErrorMessage } = useContext(TodosContext);
+
+  const visibleTodo = todos.filter(todo => {
+    switch (filterOption) {
+      case FilterOption.Active:
+        return !todo.completed;
+      case FilterOption.Completed:
+        return todo.completed;
+      case FilterOption.All:
+      default:
+        return true;
+    }
+  });
+
+  const deleteTodo = (id: number) => {
+    return todoService.deleteTodo(id)
+      .catch((error) => {
+        setErrorMessage('Unable to delete a todo');
+        throw error;
```

Comment:
do not throw the error again - it will lead to the page error, you already caught it and handled, no need to throw it again